### PR TITLE
Safari Web Extension Port from App Extension

### PR DESCRIPTION
**Pull request by Hinton**
Ports the existing Safari App Extension to the new Web Extension. Which eliminates most of the custom code required to support Safari. It also bumps the required safari version to 14.

The old xcode project was removed and a new was generated using the web extension converter.

Three features still depends on custom handlers in the Extension, but it's a fair improvement compared to before.

* Read clipboard
* Download file
* Open Popup (Only supported in safari, so we can perhaps remove this?)

I also had to add a workaround:
* StorageArea.set do not support setting a key to null in safari. Added a check which calls remove instead when value is null.

### CODE REVIEW

**Review by kspearrin — APPROVED** [review]

File: gulpfile.js [hunk 1]
```diff
@@ -186,6 +179,7 @@ function safariCopyAssets(source, dest) {
             .on('error', reject)
             .pipe(gulpif('safari/Info.plist', replace('0.0.1', manifest.version)))
```
Comment (cscharf) [inline gulpfile.js:180]: it looks like these values have been leveled in `Info.plist`... is this still valid, or do those other changes need to be reverted from the boilerplate?

File: src/safari/safari/Info.plist [hunk 2]
```diff
@@ -9,79 +9,33 @@
...
 	<string>6.0</string>
 	<key>CFBundleName</key>
-	<string>Bitwarden</string>
+	<string>$(PRODUCT_NAME)</string>
 	<key>CFBundlePackageType</key>
-	<string>XPC!</string>
+	<string>$(PRODUCT_BUNDLE_PACKAGE_TYPE)</string>
 	<key>CFBundleShortVersionString</key>
-	<string>0.0.1</string>
+	<string>1.0</string>
```
Comment (cscharf) [inline src/safari/safari/Info.plist:20]: Was this intended, noticed in the `gulpfile.js` these explicit reference strings are still there, making them no longer work.

File: src/safari/safari/Info.plist [hunk 3]
```diff
@@ -9,79 +9,33 @@
 	<key>CFBundleExecutable</key>
 	<string>$(EXECUTABLE_NAME)</string>
 	<key>CFBundleIdentifier</key>
-	<string>com.bitwarden.desktop.safari</string>
+	<string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
```
Comment (cscharf) [inline src/safari/safari/Info.plist:12]: Was this tokenization intentional?

Comment (Hinton) [reply src/safari/safari/Info.plist:12]: Yes, it will output the same.

File: src/safari/safari/Info.plist (hunk 2 above)
Comment (Hinton) [reply src/safari/safari/Info.plist:20]: I thought I had fixes this a long time ago ... Maybe I lost it after running build without `-skipcheckout`.

**Review by cscharf — APPROVED** [review]

**Pull request merged**